prime-router/src/test/kotlin/fhirengine/azure/FhirFunctionIntegrationTests.kt

```diff
@@ -0,0 +1,945 @@
+            )
+
+        val queueMessage = "{\"type\":\"route\",\"reportId\":\"${report.id}\"," +
+            "\"blobURL\":\"http://azurite:10000/devstoreaccount1/reports/receive%2Fignore.ignore-full-elr%2F" +
+            "None-${report.id}.hl7\",\"digest\":" +
+            "\"${BlobAccess.digestToString(BlobAccess.sha256Digest(routeFhirBytes))}\",\"blobSubFolderName\":" +
+            "\"ignore.ignore-full-elr\",\"schemaName\":\"\",\"topic\":\"full-elr\"}"
+
+        val fhirFunc = FHIRFunctions(
+            workflowEngine,
+
+            databaseAccess = ReportStreamTestDatabaseContainer.testDatabaseAccess
+        )
+        fhirFunc.doRoute(queueMessage, 1, fhirEngine, actionHistory)
+
+        val convertTask = ReportStreamTestDatabaseContainer.testDatabaseAccess.fetchTask(report.id)
+        assertThat(convertTask.routedAt).isNotNull()
+        ReportStreamTestDatabaseContainer.testDatabaseAccess.transact { txn ->
+            val routeTask = DSL.using(txn).select(Task.TASK.asterisk()).from(Task.TASK)
+                .where(Task.TASK.NEXT_ACTION.eq(TaskAction.translate))
+                .fetchOneInto(Task.TASK)
+            assertThat(routeTask).isNotNull()
+            val convertReportFile =
+                DSL.using(txn).select(gov.cdc.prime.router.azure.db.tables.ReportFile.REPORT_FILE.asterisk())
+                    .from(gov.cdc.prime.router.azure.db.tables.ReportFile.REPORT_FILE)
+                    .where(
+                        gov.cdc.prime.router.azure.db.tables.ReportFile.REPORT_FILE.NEXT_ACTION
+                            .eq(TaskAction.translate)
+                    )
+                    .fetchOneInto(gov.cdc.prime.router.azure.db.tables.ReportFile.REPORT_FILE)
+            assertThat(convertReportFile).isNotNull()
+        }
+        verify(exactly = 1) {
+            QueueAccess.sendMessage(elrTranslationQueueName, any())
+        }
+    }
+
+    /*
+    Send a FHIR message to an HL7v2 receiver and ensure the message receiver receives is translated to HL7v2
+     */
+    @Test
+    fun `test successfully processes a translate message when isSendOriginal is false`() {
+        // set up and seed azure blobstore
+        val blobConnectionString =
+            """DefaultEndpointsProtocol=http;AccountName=devstoreaccount1;AccountKey=keydevstoreaccount1;BlobEndpoint=http://${azuriteContainer.host}:${
+                azuriteContainer.getMappedPort(
+                    10000
+                )
+            }/devstoreaccount1;QueueEndpoint=http://${azuriteContainer.host}:${
+                azuriteContainer.getMappedPort(
+                    10001
+                )
+            }/devstoreaccount1;"""
+        val blobContainerMetadata = BlobAccess.BlobContainerMetadata(
+            "container1",
+            blobConnectionString
+        )
+
+        mockkObject(BlobAccess)
+        every { BlobAccess getProperty "defaultBlobMetadata" } returns blobContainerMetadata
```

Comment:
This is the updated mocking. Previously, the mocking was on `BlobContainerMetadata.build`, but this did not work consistently because BlobAccess.defaultMetadata is a lazy property and is only initialized the one time.